tests/infrastructure/tekton/utils.py

```diff
@@ -94,3 +106,32 @@ def process_yaml_files(file_paths, replacements, resource_kind, namespace):
         update_tekton_resources_yaml_file(file_path=file_path, replacements=replacements)
         resources.append(resource_kind(yaml_file=file_path, namespace=namespace).create())
     return resources
+
+
+def check_pipelinerun_status(
+    pipelinerun: PipelineRun, wait_timeout: int = TIMEOUT_50MIN, sleep_interval: int = TIMEOUT_1MIN
+):
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=wait_timeout,
+            sleep=sleep_interval,
+            func=lambda: pipelinerun.instance.status.conditions[0],
+        ):
+            if sample and sample["status"] != Resource.Condition.Status.UNKNOWN:
+                # There are 3 conditions.status possible: Unknown, False, True.
+                LOGGER.info(f"PipelineRun Condition : {sample}")
+                return sample
+
+    except TimeoutExpiredError:
+        LOGGER.error(
+            f"Pipelinerun: {pipelinerun.name}, "
+            f"Preparing for VM teardown due to TimeoutError. Last available conditions: {sample}"
+        )
+        raise
+
+
+def assert_pipelinerun_succeeded(pipelinerun: PipelineRun, final_status: dict[str, str]) -> None:
+    assert (
+        final_status["status"] == Resource.Condition.Status.TRUE
+        and final_status["type"] == Resource.Condition.Phase.SUCCEEDED
+    ), f"Pipelines failed to succeed. Pipeline status: {pipelinerun.instance.status}"
```

Comment:
Instead of a separate function, why not change check_pipelinerun_status() to wait for the matching condition instead?